packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3873,3 +3880,243 @@ class _CopySelectionAction extends ContextAction<CopySelectionTextIntent> {
   @override
   bool get isActionEnabled => state._value.selection.isValid && !state._value.selection.isCollapsed;
 }
+
+/// A void function that takes a [TextEditingValue].
+@visibleForTesting
+typedef TextEditingValueCallback = void Function(TextEditingValue value);
+
+/// Provides undo/redo capabilities for text editing.
+///
+/// Listens to [controller] as a [ValueNotifier] and saves relevant values for
+/// undoing/redoing. The cadence at which values are saved is a best
+/// approximation of the native behaviors of a hardware keyboard on Flutter's
+/// desktop platforms, as there are subtle differences between each of these
+/// platforms.
+///
+/// Listens for keyboard undo/redo shortcuts depending on the platform. Calls
+/// [onChanged] when a shortcut is received that would change the state of these
+/// [controller].
+@visibleForTesting
+class UndoRedo extends StatefulWidget {
```

Comment:
Are we settled on this name? I mean... have we considered something like `TextHistoryController`?

Reply:
Actually I think you're probably right... Maybe `TextEditingHistory`?